src/AppBundle/Tests/Controller/ImageGalleryControllerTest.php

```diff
@@ -0,0 +1,148 @@
+        $form = $crawler->selectButton('Opprett bildegalleri')->form();
+
+        $client->submit($this->fillForm($form));
+        $this->assertEquals(1, $client->followRedirect()->filter('h1:contains("En bra tittel")')->count());
+    }
+
+    public function testEdit()
+    {
+        $client = $this->createTeamLeaderClient();
+        $crawler = $client->request('GET', '/kontrollpanel/bildegallerier/galleri/1');
+        $form = $crawler->selectButton('Lagre endringer')->form();
+
+        $client->submit($this->fillForm($form));
+        $crawler = $client->followRedirect();
+        $this->assertEquals(1, $crawler->filter('h1:contains("En bra tittel")')->count());
+        $this->assertEquals(1, $crawler->filter('p:contains("En flott beskrivelse")')->count());
+    }
+
+    public function testUploadImage()
+    {
+        $client = $this->createTeamMemberClient();
+        $crawler = $client->request('GET', '/kontrollpanel/bildegallerier/galleri/1/ny');
+        $form = $crawler->selectButton('Lagre')->form();
+        $imageFile = tempnam(sys_get_temp_dir(), 'img');
+        imagepng(imagecreatetruecolor(1, 1), $imageFile);
+
+        $uploadedImageFile = new UploadedFile($imageFile, 'image.png', null, null, null, true);
+
+        $form['image[uploadedFile]'] = $uploadedImageFile;
+        $form['image[description]'] = 'En flott beskrivelse';
+
+        $client->submit($form);
+        $this->assertEquals(1, $client->followRedirect()->filter('img[alt="En flott beskrivelse"]')->count());
+    }
+
+    public function testDelete()
+    {
+        // Create an empty gallery
+        $client = $this->createTeamLeaderClient();
+        $crawler = $client->request('GET', '/kontrollpanel/bildegallerier');
+        $form = $crawler->selectButton('Opprett bildegalleri')->form();
+
+        $client->submit($this->fillForm($form));
+
+        // Delete this
+        $crawler = $client->followRedirect();
+        $deleteButton = $crawler->filter('.secretly-not-a-button')->first()->form();
+        $client->submit($deleteButton);
+        $crawler = $client->followRedirect();
+        $this->assertEquals(0, $crawler->filter('a:contains("En bra tittel")')->count());
+    }
+
+    public function testEditImage()
+    {
+        $client = $this->createTeamMemberClient();
+        $crawler = $client->request('GET', '/kontrollpanel/bildegallerier/rediger/1');
+        $form = $crawler->selectButton('Endre beskrivelse')->form();
+
+        $form['image[description]'] = 'En flott beskrivelse';
+
+        $client->submit($form);
+        $this->assertEquals(1, $client->followRedirect()->filter('img[alt="En flott beskrivelse"]')->count());
+    }
+
+    public function testDeleteImage()
+    {
+        $client = $this->createTeamMemberClient();
+        $crawler = $client->request('GET', '/kontrollpanel/bildegallerier/rediger/1');
+        $form = $crawler->selectButton('Slett bilde')->form();
```

Comment:
Hva skjer med bildet? Blir det borte for alltid?

Reply:
Bildet blir der :+1: